model/src/pyrenew/deterministic/deterministic.py

```diff
@@ -83,8 +90,14 @@ def sample(
         Returns
         -------
         tuple
-            Containing the stored values during construction.
+            Containing the stored values during construction wrapped in a SampledValue.
         """
         if record:
             npro.deterministic(self.name, self.vars)
-        return (self.vars,)
+        return (
+            SampledValue(
+                value=self.vars,
+                t_start=self.t_start,
+                t_unit=self.t_unit,
+            ),
+        )
```

Comment:
Noting here that this will be the gotcha of the new system. If I have 

```python
some_time_aware_sampled_value = stochastic_rv.sample()
```
and I want to pass it somewhere else downstream as a `DeterministicVariable`, I should _not_ do:

```python
my_det_var = DeterministicVariable(vars=some_time_aware_sampled_value,
                                                             name="my_det_var")
```

Instead one should do:

```python
my_det_var = DeterministicVariable(
               vars=some_time_aware_sampled_value.value,
               name="my_det_var", 
               t_start=some_time_aware_sampled_value.t_start,
               t_unit=some_time_aware_sampled_value.t_unit)
```

Reply:
Note that if we do #307 and also take my suggestion above of renaming the `vars` attribute of `DeterministicVariable` to `value`, then we could use unpacking here to turn `SampledValues` into `DeterministicVariables`:

```python
my_det_var = DeterministicVariable(
               name="my_det_var", 
              **some_time_aware_sampled_value)
```
I am on the fence RE: whether this is nicely succinct or too opaque/magic.

Reply:
In any case we should document the gotcha somewhere, perhaps.

Reply:
The current implementation checks that `vars` are indeed `ArrayLike`, which the `SampledValue` should not be, so I don't think this will cause such issues.

Reply:
I was thinking more about usage guidance than erroring, though the error is nice to have.

Reply:
Sounds good. Flag as todo in #304?